diff --git a/node_modules/abcjs/types/index.d.ts b/node_modules/abcjs/types/index.d.ts
index de64044..4473b2f 100644
--- a/node_modules/abcjs/types/index.d.ts
+++ b/node_modules/abcjs/types/index.d.ts
@@ -791,7 +791,16 @@ declare module 'abcjs' {
 		startChar: number;
 		endChar: number;
 		duration: number;
-		pitches?: Array<any>; // TODO-PER
+		pitches?: Array<{
+			accidental?: AccidentalName;
+			pitch: number;
+			name: string;
+			startSlur?: Array<{label: number}>;
+			endSlur?: Array<number>;
+			startTie?: {};
+			endTie?: boolean;
+			verticalPos: number;
+		}>;
 		rest?: { type: 'rest' | 'spacer' | 'invisible' | 'invisible-multimeasure' | 'multimeasure', text? : number};
 	}
 	export type VoiceItem = VoiceItemClef | VoiceItemBar | VoiceItemGap | VoiceItemKey | VoiceItemMeter | VoiceItemMidi | VoiceItemOverlay | VoiceItemPart | VoiceItemScale | VoiceItemStem | VoiceItemStyle | VoiceItemTempo | VoiceItemTranspose | VoiceItemNote;
@@ -935,6 +944,9 @@ declare module 'abcjs' {
 	export interface AbcElem {
 		el_type: string; //TODO enumerate these
 		abselem: AbsoluteElement;
+		averagepitch: number;
+		maxpitch: number;
+		minpitch: number;
 		beambr?: number;
 		chord?: Array<{name: string; position: ChordPlacement}>
 		decoration: Array<string> //TODO enumerate these
@@ -945,9 +957,12 @@ declare module 'abcjs' {
 		gracenotes?: Array<{duration: number; name:string; pitch: number; verticalPosition: number;}>
 		lyric?: Array<{syllable: string; divider: ' ' | '-' | '_';}>
 		noStem?: boolean
+		currentTrackMilliseconds?: number
+		currentTrackWholeNotes?: number | Array<number>
 		midiPitches?: MidiPitches;
 		midiGraceNotePitches?: MidiGracePitches;
 		pitches?: Array<{
+			accidental?: AccidentalName;
 			pitch: number;
 			name: string;
 			startSlur?: Array<{label: number}>;
