diff --git a/node_modules/abcjs/types/index.d.ts b/node_modules/abcjs/types/index.d.ts
index de64044..051032d 100644
--- a/node_modules/abcjs/types/index.d.ts
+++ b/node_modules/abcjs/types/index.d.ts
@@ -791,7 +791,16 @@ declare module 'abcjs' {
 		startChar: number;
 		endChar: number;
 		duration: number;
-		pitches?: Array<any>; // TODO-PER
+		pitches?: Array<{
+			accidental?: AccidentalName;
+			pitch: number;
+			name: string;
+			startSlur?: Array<{label: number}>;
+			endSlur?: Array<number>;
+			startTie?: {};
+			endTie?: boolean;
+			verticalPos: number;
+		}>;
 		rest?: { type: 'rest' | 'spacer' | 'invisible' | 'invisible-multimeasure' | 'multimeasure', text? : number};
 	}
 	export type VoiceItem = VoiceItemClef | VoiceItemBar | VoiceItemGap | VoiceItemKey | VoiceItemMeter | VoiceItemMidi | VoiceItemOverlay | VoiceItemPart | VoiceItemScale | VoiceItemStem | VoiceItemStyle | VoiceItemTempo | VoiceItemTranspose | VoiceItemNote;
@@ -948,6 +957,7 @@ declare module 'abcjs' {
 		midiPitches?: MidiPitches;
 		midiGraceNotePitches?: MidiGracePitches;
 		pitches?: Array<{
+			accidental?: AccidentalName;
 			pitch: number;
 			name: string;
 			startSlur?: Array<{label: number}>;
